app/services/proforma_service/convert_task_to_proforma_task.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module ProformaService
+  class ConvertTaskToProformaTask < ServiceBase
+    def initialize(task:, options: {})
+      @task = task
+      @options = options
+    end
+
+    def execute
+      create_task
+    end
+
+    private
+
+    def create_task
+      Proforma::Task.new(
+        {
+          title: @task.title,
+          description: description,
+          internal_description: @task.internal_description,
+          proglang: proglang,
+          files: @task.files.map { |file| task_file file },
+          tests: tests,
+          uuid: @task.uuid,
+          parent_uuid: @task.parent_uuid,
+          language: @task.language,
+          model_solutions: model_solutions
+        }.compact
+      )
+    end
+
+    def description
+      return @task.description if @options[:description_format] == 'md'
+
+      Kramdown::Document.new(@task.description).to_html.strip
+    end
+
+    def proglang
+      {name: @task.programming_language&.language, version: @task.programming_language&.version}
+    end
+
+    def model_solutions
+      @task.model_solutions.map do |model_solution|
+        Proforma::ModelSolution.new(
+          id: model_solution.xml_id,
+          description: model_solution.description,
+          internal_description: model_solution.internal_description,
+          files: model_solution.files.map do |file|
+            task_file(file)
+          end
+        )
+      end
+    end
+
+    def tests
+      @task.tests.map do |test|
+        file = test.files.first
+        Proforma::Test.new(
+          id: test.xml_id,
+          title: test.title,
+          description: test.description,
+          internal_description: test.internal_description,
+          test_type: test.test_type,
+          files: test.files.map { |test_file| task_file test_file },
+          configuration: test_configuration(test, file)
+        )
+      end
+    end
+
+    def test_configuration(_test, file)
+      {
+        'entry-point' => file&.full_file_name
```

Comment:
There is no testing framework / version any more as this is not required by ProFormA, I assume? And we don't need it for CodeOcean, right? Probably, that's the case; I just want to be sure :)

Reply:
Yes it just seems to be a remnant of the old Codeharbor. But it may make sense to add this again in the future.
I'm currently considering to remove the entry-point as well, because it is a custom field too.
Do we need that for Codeocean? 
I actually think we do, because a test in ProFormA can have multiple files and in that case we wouldn't know which one to execute - what do you think?

Reply:
You're right. Within CodeOcean, we require (additional) information to determine which test files can be executed. We further consider all other test files as "regular" files, i.e. non-executable files required for the test run. This could be a configuration file for a linter or some mock class / helper for a test. 

What if we store that information within [the test meta-data](https://github.com/ProFormA/proformaxml/blob/master/Whitepaper.md#the-test-meta-data-element)? According to the whitepaper, this might work?

Reply:
I just realized that the entry-point parameter wasn't arbitrarily added, but actually part of ProFormA.
It's part of this sub-xsd for unittests: https://github.com/ProFormA/proformaxml/blob/master/proforma-unittest.xsd
I originally saw it in the external files tried to import: https://github.com/ProFormA/examples/tree/master/proforma_v2.0.x/ostfalia/task
Nonetheless, since it's not part of the main proforma.xsd I would ignore that sub-xsd for now and continue with the meta-data implementation, we can add support for those subxsd's at a later point

Reply:
Hm, I see. Of course, I also didn't remember the sub-xsd (and their implications) any longer. I agree to have a look at those later